peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
+- Human-readable and structured output can be sent to separate log streams. This
+  is useful for log aggregation systems that process structured data independently
+  from human-readable data.
 
-Python Semantics in Tag Strings
--------------------------------
+(For a more detailed implementation of the custom formatters and logger, as well
+as an example of how to use dictionary-based configuration, see 
+`logging.py <https://github.com/davepeck/pep750-examples/blob/main/pep/logging.py>`_
+and `test_logging.py <https://github.com/davepeck/pep750-examples/blob/main/pep/test_logging.py>`_
+in the `pep750-examples repository <https://github.com/davepeck/pep750-examples/>`_.)
 
-Python template languages and other DSLs have semantics quite apart from Python.
-Different scope rules, different calling semantics e.g. for macros, their own
-grammar for loops, and the like.
 
-This means all tools need to write special support for each language. Even then,
-it is usually difficult to find all the possible scopes, for example to autocomplete
-values.
+Example: HTML Templating
+-------------------------
 
-However, f-strings do not have this issue. An f-string is considered part of Python.
-Expressions in curly braces behave as expected and values should resolve based on
-regular scoping rules. Tools such as mypy can see inside f-string expressions,
-but will likely never look inside a Jinja2 template.
+This PEP contains several short HTML templating examples. It turns out that the 
+"hypothetical" ``html()`` function mentioned in the  `Motivation`_ section 
+(and a few other places in this PEP) exists and is available in the 
+`pep750-examples repository <https://github.com/davepeck/pep750-examples/>`_.
+If you're thinking about building a DSL with template strings, we hope you'll
+find it useful.
 
-DSLs written with tag strings will inherit much of this value. While we can't expect
-standard tooling to understand the "domain" in the DSL, they can still inspect
-anything expressible in an f-string.
 
 Backwards Compatibility
 =======================
 
-Like f-strings, use of tag strings will be a syntactic backwards incompatibility
+Like f-strings, use of template strings will be a syntactic backwards incompatibility
 with previous versions.
 
+
 Security Implications
 =====================
 
-The security implications of working with interpolations, with respect to
+The security implications of working with template strings, with respect to
 interpolations, are as follows:
 
 1. Scope lookup is the same as f-strings (lexical scope). This model has been
    shown to work well in practice.
 
-2. Tag functions can ensure that any interpolations are done in a safe fashion,
-   including respecting the context in the target DSL.
+2. Code that processes ``Template`` instances can ensure that any interpolations 
+   are processed in a safe fashion, including respecting the context in the 
+   target DSL.
```

Comment:
Same comment as above on just DSLs vs the wider use-cases aluded to in the introduction

Reply:
Cleaned this up too. Thanks.